tests/net/gnrc_sixlowpan_frag_minfwd/main.c

```diff
@@ -624,10 +624,16 @@ static void test_minfwd_forward__ENOMEM__netif_hdr_build_fail(void)
     gnrc_pktsnip_t *pkt, *frag, *filled_space;
 
     vrbe->super.arrival = xtimer_now_usec();
+
+    /* offset: 2 * sizeof(gnrc_pktsnip_t) + movement due to mark */
+#if (__SIZEOF_POINTER__ == 8)
+        size_t offset = 155U;
+#else
+        size_t offset = 115U;
+#endif
```

Comment:
could you point me to a code / documentation line where i could see what is meant by 'mark'

Reply:
This was not my comment, I just moved it, but my understanding is that it essentially refers to this [`sys/include/net/gnrc/pktbuf.h:130`.](https://github.com/RIOT-OS/RIOT/blob/b0261f31abc35d9e33405abccd0a23e8be56494d/sys/include/net/gnrc/pktbuf.h#L97)

Reply:
Thank you for providing the hint.

sadly i am not sure about this change gnrc_pktbuf_add is usually used on network pakets/buffers  that should have a similar structure on 8 to 64Bit (since they should read /write the same).

Reply:
Maybe @miri64 can shed some light on this?

Reply:
The network buffers are still the same, but they are combined with `gnrc_pktsnip_t` and `_unused_t` elements in the "static pktbuf heap" (`_static_buf[CONFIG_GNRC_PKTBUF_SIZE]`) and all elements are aligned to the size of `_unused_t`, making it architecture dependent.
The test tries to trigger a `ENOMEM` error in `gnrc_sixlowpan_frag_minfwd_forward` due to a full buffer.
I did some testing and the state of the buffer after the two adds and the mark operation should be
```
S1|D1|S2|U|S3|D2|D3
```
with 
```
Sx: gnrc_pktsnip_t - size: _aligned(sizeof(gnrc_pktsnip_t))
Dx: data buffer - size: _aligned(bufferSize)
U: Free space that is too small for the `gnrc_sixlowpan_frag_minfwd_forward` call
```

After some calculations, the offset should be
```
size_t first_pkt_snip = _align(sizeof(gnrc_pktsnip_t));
size_t test_pkt_size = _align(sizeof(gnrc_pktsnip_t)) + _align(sizeof(_test_nth_frag));
size_t marked_pkt_size = _align(sizeof(gnrc_pktsnip_t)) + _align(sizeof(sixlowpan_frag_n_t)) +  
     _align(sizeof(_test_nth_frag) - sizeof(sixlowpan_frag_n_t));
size_t offset = first_pkt_snip + test_pkt_size + marked_pkt_size;

TEST_ASSERT_NOT_NULL((filled_space = gnrc_pktbuf_add(
        NULL, NULL,
        CONFIG_GNRC_PKTBUF_SIZE  - offset,
        GNRC_NETTYPE_UNDEF
    )));
```
However, one problem with this code is that we can't call `_align' from the test.